pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -683,23 +688,35 @@ func (r *ReconcilePerconaServerMongoDB) getRemovedSfs(ctx context.Context, cr *a
 	}
 
 	appliedRSNames := make(map[string]struct{}, len(cr.Spec.Replsets))
-	for _, v := range cr.Spec.Replsets {
-		appliedRSNames[cr.Name+"-"+v.Name] = struct{}{}
+
+	for _, rs := range cr.Spec.Replsets {
+		appliedRSNames[rs.Name] = struct{}{}
+	}
+
+	// extractRSName trims CR name and component leaving only RS name
+	// E.g. extracts 'rs1' from `my-cluster-name-rs1` or `my-cluster-name-rs1-arbiter`
+	extractRSName := func(stsName, component string) string {
+		_, cut, _ := strings.Cut(stsName, cr.Name)
+		trim := strings.Trim(cut, component)
+		return strings.Trim(trim, "-")
 	}
```

Comment:
don't we have `app.kubernetes.io/replset` label to get replset name?

Reply:
I love you, yes we have it!!

Reply:
Fixed with pleasure: https://github.com/percona/percona-server-mongodb-operator/pull/1306/commits/b312cfccc5e31bcd7fa684dc17eb52cd7397ee77